src/main/scala/io/iohk/ethereum/ledger/RewardTeller.scala

```diff
@@ -0,0 +1,69 @@
+package io.iohk.ethereum.ledger
+
+import io.iohk.ethereum.utils.MonetaryPolicyConfig
+
+/**
+  * Calculates rewards for mining blocks and ommers.
+  * Avoids floating point arithmetic. Because of that the formulas may look a bit unintuitive, but the important
+  * thing here is that we want to defer any division to be a single and final operation
+  */
+class RewardTeller(config: MonetaryPolicyConfig) {
```

Comment:
As someone familiar with it, `RewardTeller` distinguishes it, but as a first time reader it clicked in my mind when I mentally renamed it `BlockRewardCalculator`, I'd buy `RewardCalculator` either as it has context where it's created. I have some hope that `BlockRewardCalculator` might become a ledger package level trait. (Also, the first word of the description is `Calculates...`)